### PR TITLE
Describe domain section in root README

### PR DESCRIPTION
**Pull request by meaboutsoftware**
This PR includes the description of domain section and its subchapters:

- Overview - explaining that we selected Fitness Studio business domain
- Subdomains - a split into Passes, Contracts, Reports and Offers subdomains
- Potential Architecture Patterns - active record, transaction script, domain model

### CODE REVIEW

File: README.adoc [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+image::Assets/subdomains_processes.jpg[]
+
+There are 6 different processes in 4 different subdomains. There is a problem here - these subdomains do not communicate with each other and this is one of the most common problems we have in our applications. So we decided to complicate things a bit and add some communication:
+
+image::Assets//subdomains_communication.jpg[]
+
+There are 2 triggers:
+
+- when the contract is signed by you (a customer), then the new pass is registered to allow you to enter the fitness studio
+- when the contract expires, then the new offer is prepared (which will be sent to you as a customer)
```
Comment (kamilbaczek) [inline README.adoc:54]: Currently when pass expire then new offer is preapred.

File: README.adoc [hunk 2]
```diff
@@ -0,0 +1,130 @@
+# Evolutionary Architecture
+
+## Objective
+
+### Problem
```
Comment (kamilbaczek) [inline README.adoc:5]: In main readme i think is worth to mention about things that are out of scope repository. 

Like:
Frontend 
Deployment
Authorization
etc

I'm suggesting to add Header now or in the future

File: README.adoc [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+### Chapter 4: Domain Model
+
+## Repository Structure
+
+### Overview
+
+### How To Navigate?
+
+## Technologies and Tools Used
+
```
Comment (kamilbaczek) [inline README.adoc:129]: In main readme i think is worth to mention about things that are out of scope repository. 

Like:
Frontend 
Deployment
Authorization
etc

I'm suggesting to add Header now or in the future

**Comment by kamilbaczek** [comment]
Hi, you did amazing job with this description. It is looks professional but it easy to read 🔥

File: README.adoc (hunk 3 above)
Comment (meaboutsoftware) [reply README.adoc:129]: Good point! I will add it when writing section about technologies and tools used :)

File: README.adoc (hunk 1 above)
Comment (meaboutsoftware) [reply README.adoc]: Done!

**Review by kamilbaczek — APPROVED** [review]

**Pull request merged**